### PR TITLE
[ROCm] Initial ir.Scan/aten.cumsum lowering support on ROCm

### PR DESCRIPTION
**Pull request by jataylo**
It was noted in https://github.com/pytorch/pytorch/pull/117992 that ROCm is still falling back to eager with scan's with inductor.

Initially as part of https://github.com/pytorch/pytorch/pull/106581 ROCm was disabled on this feature due to lack of triton support.

This PR will enable support for lowering scan operations on ROCm.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by peterbell10 — APPROVED** [review]

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -364,13 +364,12 @@ def wrapper_noop_set_seed(op, *args, **kwargs):
...
-    inductor_override_kwargs.update(
-        {
-            # We have better precision than eager
-            ("cumsum", "cuda", f16): {"reference_in_float": True},
-        }
-    )
+inductor_override_kwargs.update(
+    {
+        # We have better precision than eager
+        ("cumsum", "cuda", f16): {"reference_in_float": True},
```
[the remark this answers is not in the document]
Comment (jataylo) [reply test/inductor/test_torchinductor_opinfo.py]: Missed that, updated now!

**Comment by jataylo** [comment]
Green last run, hopefully last conflicts before we merge this

**Pull request closed without merging**